rewrite-core/src/main/java/org/openrewrite/rpc/request/GetRef.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.rpc.request;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.openrewrite.rpc.RpcObjectData;
+import org.openrewrite.rpc.RpcSendQueue;
+
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.openrewrite.rpc.RpcObjectData.State.DELETE;
+import static org.openrewrite.rpc.RpcObjectData.State.END_OF_OBJECT;
+
+@Value
+public class GetRef implements RpcRequest {
+    int ref;
+
+    @RequiredArgsConstructor
+    public static class Handler extends io.moderne.jsonrpc.JsonRpcMethod<GetRef> {
+        private static final ExecutorService forkJoin = ForkJoinPool.commonPool();
+
+        private final Map<Integer, Object> remoteRefs;
+        private final IdentityHashMap<Object, Integer> localRefs;
+        private final AtomicInteger batchSize;
+        private final AtomicBoolean trace;
+
+        private final Map<Integer, BlockingQueue<List<RpcObjectData>>> inProgress = new ConcurrentHashMap<>();
+
```

Comment:
So is the idea here that one side or the other of an RPC setup owns the reference to a particular object and that can be used to figure out if it is safe to evict it from a cache?

Reply:
The idea is that either party should be allowed to cache both "objects" and "refs" (which ultimately might converge) without needing to coordinate the cache eviction with the other party. When a "ref" is unknown, it will then just send a "GetRef" request.